rdf/shared/src/main/scala/org/w3/banana/io/RDFWriter.scala

```diff
@@ -5,10 +5,12 @@ import java.io.OutputStream
 
 trait RDFWriter[Rdf <: RDF, M[_], +T] extends Writer[Rdf#Graph,M,T] {
 
+  def write(graph: Rdf#Graph, os: OutputStream, base: String): M[Unit] = write(graph, os, base, Nil:_*)
 
-  def write(graph: Rdf#Graph, os: OutputStream, base: String): M[Unit]
+  def write(graph: Rdf#Graph, os: OutputStream, base: String, prefixes: Prefix[Rdf]*): M[Unit]
```

Comment:
I don't think using _varargs_ for the prefixes is a good idea as:
1. it makes extending the function with any other arguments very difficult later. And it is clear that the method will need extending: for example currently it uses an `OutputStream` but there is no way to specify the encoding needed. When writing an HTTP server that has to be specified somewhere for reasons of content negotiation.  
2. Furthermore, it is quite likely that anyone who does use a number of prefixes, will have his favorite prefix library, and will be using quite a number of them. For them it will be quite natural to creating a function, or even a macro to build up a `Set` of `Prefix`es from a graph, or even from some DSL that builds graphs. 
3. Writing out prefixes one by one if there are much more than one is going to be quite ugly.

I would suggest having a `Set[Prefix]` argument as the default and setting its default value (if possible) to the empty set. Ie. `prefixes: Set[Prefix[Rdf]] = Set()` .